packages/core/echo/echo-pipeline/src/db-host/query-state.ts

```diff
@@ -181,21 +181,21 @@ export class QueryState extends Resource {
 
 // TODO(burdon): Process Filter DSL.
 const filterToIndexQuery = (filter: FilterProto): IndexQuery => {
-  invariant(!(filter.type && (filter.or ?? []).length > 0), 'Cannot mix type and or filters.');
+  invariant(!((filter.type ?? []).length > 0 && (filter.or ?? []).length > 0), 'Cannot mix type and or filters.');
```

Comment:
```suggestion
  const { type = [], or = [] } = filter;
  invariant(!(type.length > 0 && or.length > 0), 'Cannot mix type and or filters.');
```
fallback once rather than on every access